src/api/controllers/authController.js

```diff
@@ -97,4 +99,201 @@ export default class Auth {
       return sendError(status.SERVER_ERROR, res, 'logout', 'Please try again, Thanks');
     }
   }
+
+  /**
+   * This function gets all the users that are stored in the User table
+   *
+   * @static
+   * @param {Object} req the request
+   * @param {Object} res the response to be sent
+   * @memberof users
+   * @returns {Object} res
+   */
+  static async getUsers(req, res) {
+    // default constants
+    const defaultLimit = 10;
+    const defaultOffset = 0;
+    const result = {};
+    const emptyUserArray = 0;
+
+    // const page = req.query.page || 1;
+    const limit = req.query.limit || defaultLimit;
+    const offset = req.query.offset || defaultOffset;
+    const users = await User.findAndCountAll({
+      offset,
+      limit
+    });
+
+    result.status = 200;
+    result.pagedUsers = paginateUser(users.count, users.rows, offset, limit);
+    result.users = users.rows;
+    if (users.rows.length > emptyUserArray) {
+      res.status(status.OK).json({
+        result
+      });
+    } else {
+      res.status(status.NOT_FOUND).json({
+        status: 404,
+        message: 'There no more users at the moment.'
+      });
+    }
+  }
+
+  /**
+   * Update User
+   *
+   * @static
+   * @param {object} req - request body
+   * @param {object} res - response body
+   * @returns { object } - response
+   * @memberof Auth
+   */
+  static async updateUser(req, res) {
+    const { email } = req.params;
+    const {
+      firstName, lastName, username, password
+    } = req.body;
+
+    const updateContent = {
+      email,
+      firstName,
+      lastName,
+      username,
+      password
+    };
+
+    const result = await User.update({
+      email,
+      firstName: updateContent.firstName,
+      lastName: updateContent.lastName,
+      username: updateContent.username,
+      password: updateContent.password
+    },
+    {
+      where: {
+        email
+      }
+    });
+    return res.status(status.OK).json({
+      status: status.OK,
+      message: 'user updated successufully',
+      data: result
+    });
+  }
+
+  /**
+   * make User a modulator
+   *
+   * @static
+   * @param {object} req - request body
+   * @param {object} res - response body
+   * @returns { object } - response
+   * @memberof Auth
+   */
+  static async makeModulator(req, res) {
+    const { email } = req.params;
+    const role = 'modulator';
+    const updateRole = role;
+
```

Comment:
what the use of updateRole? yet you can use role in your query
I don't find the reason for passing the role through another variable just to change the name

Reply:
@alainburindi I resolved this